app/controllers/user/omniauth_callbacks_controller.rb

```diff
@@ -27,7 +27,9 @@ def sign_in_with_facebook
                         kind: t('user.omniauth_callbacks.facebook.kind')) if is_navigational_format?
     else
       session['devise.facebook_data'] = auth
-      redirect_to new_user_registration_path
+      redirect_to new_user_registration_path,
+                  danger: t('user.omniauth_callbacks.facebook.sign_in_failure',
+                            errors: @user.errors.full_messages.to_sentence)
```

Comment:
final stickler: check you check all the failure strings -- do we use errors or error?